vscode/src/chat/chat-view/ChatPanelsManager.ts

```diff
@@ -279,10 +281,12 @@ export class ChatPanelsManager implements vscode.Disposable {
         try {
             logDebug('ChatPanelsManager', 'restorePanel')
             // Panel already exists, just reveal it
-            const provider = this.panelProvidersMap.get(chatID)
+            const provider = this.panelProviders.find(p => p.sessionID === chatID)
             if (provider) {
-                provider.webviewPanel?.reveal()
-                return provider
+                if (provider.sessionID === chatID) {
```

Comment:
```suggestion
                if (provider?.sessionID === chatID) {
```
so we can remove the `if (provider) {` above?